### PR TITLE
*: introduce slog_panic and SlogFormat

### PR DESCRIPTION
**Pull request by BusyJay**
### What is changed and how it works?
Issue Number: Ref #12842 

What's Changed:

```commit-message
These two are helpers to utilize the static KV pairs in logger. In the
past, we use `logger.list()` to try to format the configured KV pairs,
but it will not work as values are omitted.
```

### Check List 

Tests 

- Unit test

### Release note 

```release-note
None
```

### CODE REVIEW

**Review by tabokie — APPROVED** [review]

File: components/raftstore-v2/src/operation/command/admin/split.rs [hunk 1]
```diff
@@ -326,10 +327,10 @@ impl<EK: KvEngine, R: ApplyResReporter> Apply<EK, R> {
         // We will freeze the memtable rather than flush it in the following PR.
         let tablet = self.tablet().clone();
         let mut checkpointer = tablet.new_checkpointer().unwrap_or_else(|e| {
-            panic!(
-                "{:?} fails to create checkpoint object: {:?}",
-                self.logger.list(),
-                e
+            slog_panic!(
+                self.logger,
+                "ails to create checkpoint object";
```
Comment (ethercflow) [inline components/raftstore-v2/src/operation/command/admin/split.rs:332]: ails to fails?

File: components/tikv_util/src/log.rs [hunk 2]
```diff
@@ -82,3 +82,145 @@ macro_rules! debug(($($args:tt)+) => {
...
+    };
+    let record = slog::record_static!(slog::Level::Trace, "");
+    let args = format_args!("");
+    let record = Record::new(&record, &args, slog::b!());
+    // Serialize borrow list first to make region_id, peer_id at the end.
+    borrow_list.serialize(&record, &mut formatter).unwrap();
+    kv_list.serialize(&record, &mut formatter).unwrap();
+}
+
+/// A helper macros to panic with the key-value list of a slog logger.
```
Comment (ethercflow) [inline components/tikv_util/src/log.rs:145]: macro?

**Review by ethercflow — APPROVED** [review]
Rest LGTM

**Pull request merged**